File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
    
        string s="";
      int len = chars.size();
      
      if(len==1)return 1;
      
      for(int i=0; i<len; i++){
        char c = chars[i];
        int coun = 0;
        
        while(i < len && chars[i]==c){
        coun ++;
          i++;
        }
        i--;
        
        s += c;
        if(coun>1){
          s += to_string(coun);
        }
      }
      for(int i=0; i<s.size(); i++){
        chars[i] = s[i];
      }
      return s.size();
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        StringBuilder st = new StringBuilder();
        
        int counter =0, j=0;
        for(int i=0; i<chars.length; i++){
            chars[j++] = chars[i];
            counter=1;
            while(i+1<chars.length && chars[i] == chars[i+1]){
                counter++;
                i++;
            }
            if(counter>1){
                String ss = new String(Integer.toString(counter));
                for(int k=0; k<ss.length() ;k++){
                  chars[j++] = ss.charAt(k);
                }
            }
        }
        return j;
    }
}